[CI] EsqlAsyncSecurityIT testLimitedPrivilege failing

**Build Scans:**
- [elasticsearch-periodic #4166 / openjdk21_checkpart4_java-matrix](https://gradle-enterprise.elastic.co/s/fqzqfg3nncmvw)
- [elasticsearch-pull-request #32154 / part-4](https://gradle-enterprise.elastic.co/s/gbp623njpkux2)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:security:javaRestTest" --tests "org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT.testLimitedPrivilege" -Dtests.seed=134C280A2EA3C3FE -Dtests.locale=tg-TJ -Dtests.timezone=Pacific/Galapagos -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testLimitedPrivilege'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: a string containing "unauthorized for user [test-admin] run as [metadata1_read2] with effective roles [metadata1_read2] on indices [index-user1]"
     but: was "{"error":{"root_cause":[{"type":"no_shard_available_action_exception","reason":"No shard available for [get [.async-search][TiGvZNvmRTaoKLh-3UCoxA]: routing [null]]","stack_trace":"org.elasticsearch.action.NoShardAvailableActionException: No shard available for [get [.async-search][TiGvZNvmRTaoKLh-3UCoxA]: routing [null]]\n"}],"type":"no_shard_available_action_exception","reason":"No shard available for [get [.async-search][TiGvZNvmRTaoKLh-3UCoxA]: routing [null]]","stack_trace":"org.elasticsearch.action.NoShardAvailableActionException: No shard available for [get [.async-search][TiGvZNvmRTaoKLh-3UCoxA]: routing [null]]\n"},"status":503}"
```

**Issue Reasons:**
- [main] 2 failures in test testLimitedPrivilege (0.4% fail rate in 530 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -153,9 +153,6 @@ tests:
 - class: org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT
   method: test {p0=search/500_date_range/from, to, include_lower, include_upper deprecated}
   issue: https://github.com/elastic/elasticsearch/pull/113286
-- class: org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT
-  method: testLimitedPrivilege
-  issue: https://github.com/elastic/elasticsearch/issues/113419
 - class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
   method: test {categorize.Categorize}
   issue: https://github.com/elastic/elasticsearch/issues/113428

--- a/x-pack/plugin/esql/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/esql/EsqlAsyncSecurityIT.java
+++ b/x-pack/plugin/esql/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/esql/EsqlAsyncSecurityIT.java
@@ -154,6 +154,14 @@ public class EsqlAsyncSecurityIT extends EsqlSecurityIT {
                     } catch (InterruptedException ex) {
                         throw new RuntimeException(ex);
                     }
+                } else if (statusCode == 503 && message.contains("No shard available for [get [.async-search]")) {
+                    // Workaround for https://github.com/elastic/elasticsearch/issues/113419
+                    logger.warn(".async-search index shards not yet available", e);
+                    try {
+                        Thread.sleep(500);
+                    } catch (InterruptedException ex) {
+                        throw new RuntimeException(ex);
+                    }
                 } else {
                     throw e;
                 }

